packages/office-ui-fabric-react/src/components/MarqueeSelection/examples/MarqueeSelection.Basic.Example.tsx

```diff
@@ -48,73 +48,52 @@ const styles = mergeStyleSets({
       },
     },
   },
+  checkbox: {
+    margin: '10px 0',
+  },
 });
 
-const checkboxStyles: Partial<ICheckboxStyles> = { root: { margin: '10px 0' } };
-
-export interface IMarqueeSelectionBasicExampleState {
-  isMarqueeEnabled: boolean;
-}
-
-export class MarqueeSelectionBasicExample extends React.Component<{}, IMarqueeSelectionBasicExampleState> {
-  private _selection: Selection;
-  private _isMounted: boolean;
-
-  constructor(props: {}) {
-    super(props);
-
-    this.state = {
-      isMarqueeEnabled: true,
-    };
-
-    this._selection = new Selection({
-      onSelectionChanged: () => {
-        if (this._isMounted) {
-          this.forceUpdate();
-        }
-      },
+const useForceUpdate = () => {
+  const [, setCount] = React.useState(0);
+  console.log('update');
+  return React.useCallback(() => setCount(current => current + 1), []);
+};
+
+export const MarqueeSelectionBasicExample: React.FunctionComponent = () => {
+  const [isMarqueeEnabled, { toggle: toggleIsMarqueeEnabled }] = useBoolean(true);
+  const forceUpdate = useForceUpdate();
+  const [selection] = React.useState(() => {
+    const newSelection = new Selection<IPhoto>({
+      items: PHOTOS,
+      onSelectionChanged: forceUpdate,
     });
 
-    this._selection.setItems(PHOTOS);
-  }
-
-  public componentDidMount(): void {
-    this._isMounted = true;
-  }
-
-  public render(): JSX.Element {
-    return (
-      <MarqueeSelection selection={this._selection} isEnabled={this.state.isMarqueeEnabled}>
-        <Checkbox styles={checkboxStyles} label="Is marquee enabled" defaultChecked={true} onChange={this._onChange} />
-        <p>Drag a rectangle around the items below to select them:</p>
-        <ul className={styles.photoList}>
-          {PHOTOS.map((photo, index) => (
-            <div
-              key={index}
-              className={css(styles.photoCell, this._selection.isIndexSelected(index) && 'is-selected')}
-              data-is-focusable={true}
-              data-selection-index={index}
-              onClick={this._log('clicked')}
-              style={{ width: photo.width, height: photo.height }}
-            >
-              {index}
-            </div>
-          ))}
-        </ul>
-      </MarqueeSelection>
-    );
-  }
-
-  private _log(text: string): () => void {
-    return (): void => {
-      console.log(text);
-    };
-  }
-
-  private _onChange = (
-    ev: React.FormEvent<HTMLElement | HTMLInputElement>,
-    isMarqueeEnabled: boolean | undefined,
-  ): void => {
-    this.setState({ isMarqueeEnabled: isMarqueeEnabled! });
-  };
-}
+    newSelection.setItems(PHOTOS);
```

Comment:
You should be able to get rid of the setItems call now that you added items as a constructor prop?